pyquil/reference_simulator.py

```diff
@@ -163,8 +177,37 @@ class ReferenceDensitySimulator(AbstractQuantumSimulator):
     def __init__(self, n_qubits: int, rs: RandomState = None):
         self.n_qubits = n_qubits
         self.rs = rs
-        self.density = np.zeros((2 ** n_qubits, 2 ** n_qubits), dtype=np.complex128)
-        self.density[0, 0] = complex(1.0, 0)
+        self.initial_density = None
+        # set self.density via
+        self.reset()
+
+    def set_initial_state(self, state_matrix):
+        """
+        This method is the correct way (TM) to set the state matrix to a state other than the
+        default state. The default state of ReferenceDensitySimulator is |000...00>.
```

Comment:
This docstring doesn't explain that the initial state for all future resets will be updated. 

It would make more sense to me if we removed ```self.reset()``` (line 209) and changed these two sentences 
```
This method is the correct way (TM) to update the initial state matrix that is initialized every time reset() is called. The default initial state of ReferenceDensitySimulator is |000...00>.

Note that the current state matrix is not affected by this method; you must change it directly or else call reset() after calling this method. 

To restore default state initialization behavior of ReferenceDensitySimulator pass in ``state_matrix = None.`` 
```

Reply:
- [x] todo